main.cpp

```diff
@@ -175,6 +232,22 @@ struct Dog
     //5) breed (std::string)
     std::string breed;
 
+    float cutHairToNewLength(float newLengthInCm)
+    {
+        float amountCutOff = 0.0f;
+        if(newLengthInCm < hairLengthInCM)
+        {
+            amountCutOff = hairLengthInCM - newLengthInCm;
+            std::cout << "amount cut off:" << amountCutOff << std::endl;
+            while(amountCutOff > 0)    
+            {
+                bark();
+                amountCutOff--;
```

Comment:
use `pre-decrement/increment`, not `post-decrement/increment`
See https://en.cppreference.com/w/cpp/language/operator_incdec 
- [ ] revise